modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,408 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background with that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ *
+ * @return array $metadata
+ */
+function wp_dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+
+		return $metadata;
+	}
+	$dominant_color = wp_get_dominant_color( $attachment_id );
+
+	if ( ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	return $metadata;
+}
+add_filter( 'wp_generate_attachment_metadata', 'wp_dominant_color_metadata', 10, 2 );
+
+
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @param array $metadata Metadata for the attachment.
+ * @param int   $attachment_id attachement id.
+ *
+ * @return array $metadata
+ */
+function wp_has_transparency_metadata( $metadata, $attachment_id ) {
+
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+
+		return $metadata;
+	}
+
+	$has_transparency = wp_get_has_transparency( $attachment_id );
+
+	if ( ! empty( $has_transparency ) ) {
+		$metadata['has_transparency'] = $has_transparency;
+	}
+
+	return $metadata;
+}
+add_filter( 'wp_generate_attachment_metadata', 'wp_has_transparency_metadata', 10, 2 );
```

Comment:
I think it would be more efficient to make this one function that handles both of these. Having two separate filters with some duplicate code here should be unnecessary.